flask/app/datasets.py

```diff
@@ -208,6 +197,20 @@ def list_datasets(page: int, limit: int) -> Response:
         for dataset in datasets
     ]
 
+    app.logger.debug(
+        "%d datasets to be returned; %d limit; %d total_count",
+        len(datasets),
+        limit or -1,
+        total_count,
+    )
+
+    if len(datasets) != min(limit or total_count, total_count):
+        app.logger.warning(
+            "Datasets to be returned is incorrect; expected: '%d', actual: '%d'",
+            min(limit or total_count, total_count),
+            len(datasets),
+        )
+
```

Comment:
I don't think we need the warning b/c our tests should fail at merge time if there's an issue here. Is the debugging statement for development purposes or logging? (TBH, I'm still not 100% familiar with logging strategy).